<!DOCTYPE html>

<html class="writer-html5" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="IE=edge" http-equiv="X-UA-Compatible"/>
<meta content="width=device-width, initial-scale=1.0" name="viewport"/><meta content="CIDR" name="author"/><link href="https://CIDR-network-hub.github.io/analysis_landing/" rel="canonical"/>
<link href="../img/favicon.ico" rel="shortcut icon"/>
<title>Analysing data - CIDR Metagenomics</title>
<link href="../css/theme.css" rel="stylesheet"/>
<link href="../css/theme_extra.css" rel="stylesheet"/>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" rel="stylesheet"/>
<link href="../assets/_mkdocstrings.css" rel="stylesheet"/>
<script>
        // Current page data
        var mkdocs_page_name = "Analysing data";
        var mkdocs_page_input_path = "analysis_landing.md";
        var mkdocs_page_url = "/analysis_landing/";
      </script>
<!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
<script>hljs.highlightAll();</script>
<link href="../cidr_metagenomics.pdf" rel="alternate" title="PDF" type="application/pdf"/> <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }</style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
<body class="wy-body-for-nav" role="document">
<div class="wy-grid-for-nav">
<nav class="wy-nav-side stickynav" data-toggle="wy-nav-shift">
<div class="wy-side-scroll">
<div class="wy-side-nav-search">
<a class="icon icon-home" href=".."> CIDR Metagenomics
        </a>
</div>
<div aria-label="Navigation menu" class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation">
<ul>
<li class="toctree-l1"><a class="reference internal" href="..">Introduction</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Lab Resources</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../lab_protocols/">Lab protocols</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Bioinformatics</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_outline/">Outline</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_install/">Installation</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../running_metagenomics_workflow/">Starting the metagenomics workflow</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../running_organism_query/">Query a classification</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../alternative_deployment/">Deployment alternatives</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../clinical_interpretation/">Clinical interpretation</a>
</li>
</ul>
</div>
</div>
</nav>
<section class="wy-nav-content-wrap" data-toggle="wy-nav-shift">
<nav aria-label="Mobile navigation menu" class="wy-nav-top" role="navigation">
<i class="fa fa-bars" data-toggle="wy-nav-top"></i>
<a href="..">CIDR Metagenomics</a>
</nav>
<div class="wy-nav-content">
<div class="rst-content"><div aria-label="breadcrumbs navigation" role="navigation">
<ul class="wy-breadcrumbs">
<li><a aria-label="Docs" class="icon icon-home" href=".."></a></li>
<li class="breadcrumb-item active">Analysing data</li>
<li class="wy-breadcrumbs-aside">
<a class="icon icon-github" href="https://github.com/GSTT-CIDR/network_hub/edit/master/docs/analysis_landing.md"> Edit on GitHub</a>
</li>
</ul>
<hr/>
</div>
<div class="document" itemscope="itemscope" itemtype="http://schema.org/Article" role="main">
<div class="section" itemprop="articleBody">
<h1 id="analysing-data">Analysing data</h1>
<h2 id="overview">Overview</h2>
<p>Once data and metadata have been ingested into the Onyx database, you
can query it using the Onyx client, which provides a command line interface (CLI)
and Python API.  This short example
demonstrates a few principal functions.  More are described in the
<a href="https://climb-tre.github.io/onyx-client/"><code>onyx-client</code> documentation</a>.</p>
<p>This guide also assumes that you're using a Notebook Server on CLIMB,
so that once installed, the Onyx client will automatically be configured.</p>
<h2 id="onyx-client-basics">Onyx client basics</h2>
<p>First, let's install the Onyx client, which is available through the
<a href="https://pypi.org/project/climb-onyx-client">Python package</a>
<code>climb-onyx-client</code> and can thus be installed
with <code>pip</code>.  As advised in the <a href="https://docs.climb.ac.uk/notebook-servers/installing-software-with-conda/">CLIMB docs on installing
software</a>,
you should install the client in a new Conda environment.
I'll name my environment <code>onyx</code> and install <code>ipykernel</code> too
so that it's available in my Jupyter Notebooks.
<div class="language-text highlight"><pre><span></span><code><span id="__span-0-1"><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>jovyan:~$ conda create -n onyx ipykernel
</span></code></pre></div>
Let's activate this environment and install the Onyx client.
<div class="language-text highlight"><pre><span></span><code><span id="__span-1-1"><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>jovyan:~$ conda activate onyx
</span><span id="__span-1-2"><a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>(onyx) jovyan:~$ pip install climb-onyx-client
</span></code></pre></div>
On Bryn's Notebook Servers, the client will automatically be configured.
Try running the command-line client with
<div class="language-text highlight"><pre><span></span><code><span id="__span-2-1"><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>(onyx) jovyan:~$ onyx
</span></code></pre></div>
This should show you some options and commands that are available.
Have a look at your own profile with
<div class="language-text highlight"><pre><span></span><code><span id="__span-3-1"><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>(onyx) jovyan:~$ onyx profile
</span></code></pre></div>
and which projects you have access to with
<div class="language-text highlight"><pre><span></span><code><span id="__span-4-1"><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>(onyx) jovyan:~$ onyx projects
</span></code></pre></div>
You should see <code>mscapetest</code> listed.</p>
<h2 id="querying-data">Querying data</h2>
<p>As an example task, we'll see if we can find any sequencing data performed
for ZymoBIOMICS sources.  These are designed with 
<a href="https://files.zymoresearch.com/protocols/_d6300_zymobiomics_microbial_community_standard.pdf">a particular specification</a>
of DNA from eight bacteria and two yeasts.  We can use these to see if our protocol
correctly recovers the DNA fractions. I.e. if our protocol is biased.</p>
<p>From the command line, the main route to querying Onyx is via the <code>filter</code> command.
On its own, this queries the database with <em>no</em> filters.  The command
<div class="language-text highlight"><pre><span></span><code><span id="__span-5-1"><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>(onyx) jovyan:~$ onyx filter mscapetest
</span></code></pre></div>
will produce tens of thousands of lines of JSON, so let's not
do that just yet.  To first see which fields are available in the database,
we can use
<div class="language-text highlight"><pre><span></span><code><span id="__span-6-1"><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>(onyx) jovyan:~$ onyx fields mscapetest
</span><span id="__span-6-2"><a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a>...
</span><span id="__span-6-3"><a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a>├────────────────────────────────┼──────────┼───────────────────┼──────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────┤
</span><span id="__span-6-4"><a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a>│ extraction_enrichment_protocol │ optional │ text              │ Details of nucleic acid extraction and optional enrichment steps.            │                                                                             │
</span><span id="__span-6-5"><a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a>├────────────────────────────────┼──────────┼───────────────────┼──────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────┤
</span><span id="__span-6-6"><a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a>...
</span></code></pre></div>
Let's search the database for entries with <code>zymo</code> (case-insensitive) in this field.
<div class="language-text highlight"><pre><span></span><code><span id="__span-7-1"><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>(onyx) jovyan:~$ onyx filter mscapetest --field extraction_enrichment_protocol.icontains=zymo
</span><span id="__span-7-2"><a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>...
</span></code></pre></div>
That should return JSON data for a few entries.  You may wish to format the
data as CSV or TSV with <code>--format csv</code> or <code>--format tsv</code>, respectively.</p>
<h2 id="inspecting-some-pipeline-output-on-the-command-line">Inspecting some pipeline output on the command line</h2>
<p>When data is ingested into Onyx, a taxonomic classification is automatically run.
The last part of the JSON data is usually some of this, in JSON format.
The complete reports can be found in the S3 buckets given in the
<code>'taxon_report'</code> field.  You can find this in the output you've already produced
or modify the <code>filter</code> command to only request them using the <code>--include</code> flag. e.g.
<div class="language-text highlight"><pre><span></span><code><span id="__span-8-1"><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a>(onyx) jovyan:~$ onyx filter mscapetest --field extraction_enrichment_protocol.icontains=zymo --include=taxon_reports
</span><span id="__span-8-2"><a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a>[
</span><span id="__span-8-3"><a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a>    {
</span><span id="__span-8-4"><a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a>        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-FDE50853AD/"
</span><span id="__span-8-5"><a href="#__codelineno-8-5" id="__codelineno-8-5" name="__codelineno-8-5"></a>    },
</span><span id="__span-8-6"><a href="#__codelineno-8-6" id="__codelineno-8-6" name="__codelineno-8-6"></a>    {
</span><span id="__span-8-7"><a href="#__codelineno-8-7" id="__codelineno-8-7" name="__codelineno-8-7"></a>        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-04F4495068/"
</span><span id="__span-8-8"><a href="#__codelineno-8-8" id="__codelineno-8-8" name="__codelineno-8-8"></a>    }
</span><span id="__span-8-9"><a href="#__codelineno-8-9" id="__codelineno-8-9" name="__codelineno-8-9"></a>]
</span></code></pre></div>
If you'd like multiple fields, enter each one with another <code>--include</code> flag,
or use bash's <a href="https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html">brace expansion</a>, e.g.
<div class="language-text highlight"><pre><span></span><code><span id="__span-9-1"><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a>(onyx) jovyan:~$ onyx filter mscapetest --field extraction_enrichment_protocol.icontains=zymo --include={cid,taxon_reports}
</span><span id="__span-9-2"><a href="#__codelineno-9-2" id="__codelineno-9-2" name="__codelineno-9-2"></a>[
</span><span id="__span-9-3"><a href="#__codelineno-9-3" id="__codelineno-9-3" name="__codelineno-9-3"></a>    {
</span><span id="__span-9-4"><a href="#__codelineno-9-4" id="__codelineno-9-4" name="__codelineno-9-4"></a>        "cid": "C-FDE50853AD",
</span><span id="__span-9-5"><a href="#__codelineno-9-5" id="__codelineno-9-5" name="__codelineno-9-5"></a>        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-FDE50853AD/"
</span><span id="__span-9-6"><a href="#__codelineno-9-6" id="__codelineno-9-6" name="__codelineno-9-6"></a>    },
</span><span id="__span-9-7"><a href="#__codelineno-9-7" id="__codelineno-9-7" name="__codelineno-9-7"></a>    {
</span><span id="__span-9-8"><a href="#__codelineno-9-8" id="__codelineno-9-8" name="__codelineno-9-8"></a>        "cid": "C-04F4495068",
</span><span id="__span-9-9"><a href="#__codelineno-9-9" id="__codelineno-9-9" name="__codelineno-9-9"></a>        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-04F4495068/"
</span><span id="__span-9-10"><a href="#__codelineno-9-10" id="__codelineno-9-10" name="__codelineno-9-10"></a>    }
</span><span id="__span-9-11"><a href="#__codelineno-9-11" id="__codelineno-9-11" name="__codelineno-9-11"></a>]
</span></code></pre></div>
You can conversely exclude individual fields using the <code>--exclude</code>
flag in the same way.</p>
<p>Either way, you now have the location of the taxonomy reports.  Let's have a look
with <code>s3cmd</code>.
<div class="language-text highlight"><pre><span></span><code><span id="__span-10-1"><a href="#__codelineno-10-1" id="__codelineno-10-1" name="__codelineno-10-1"></a>(onyx) jovyan:~$ s3cmd ls s3://mscapetest-published-taxon-reports/C-FDE50853AD/
</span><span id="__span-10-2"><a href="#__codelineno-10-2" id="__codelineno-10-2" name="__codelineno-10-2"></a>2023-11-10 12:56   146K  s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken.json
</span><span id="__span-10-3"><a href="#__codelineno-10-3" id="__codelineno-10-3" name="__codelineno-10-3"></a>2023-11-10 12:56     2G  s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_assignments.tsv
</span><span id="__span-10-4"><a href="#__codelineno-10-4" id="__codelineno-10-4" name="__codelineno-10-4"></a>2023-11-10 12:56   193K  s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_report.txt
</span></code></pre></div>
The plain text report is what we're after, so let's download that with <code>s3cmd</code>:
<div class="language-text highlight"><pre><span></span><code><span id="__span-11-1"><a href="#__codelineno-11-1" id="__codelineno-11-1" name="__codelineno-11-1"></a>(onyx) jovyan:~$ s3cmd get s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_report.txt
</span><span id="__span-11-2"><a href="#__codelineno-11-2" id="__codelineno-11-2" name="__codelineno-11-2"></a>download: 's3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_report.txt' -&gt; './PlusPF.kraken_report.txt'  [1 of 1]
</span><span id="__span-11-3"><a href="#__codelineno-11-3" id="__codelineno-11-3" name="__codelineno-11-3"></a> 197750 of 197750   100% in    0s     3.79 MB/s  done
</span></code></pre></div></p>
<p>If you've never seen one of these reports before, it's worth having a
quick look with a tool like <code>less</code> or by opening it using the
JupyterLab file browser.  For reference, it's worth showing the header
<div class="language-text highlight"><pre><span></span><code><span id="__span-12-1"><a href="#__codelineno-12-1" id="__codelineno-12-1" name="__codelineno-12-1"></a>(onyx) jovyan:~$ head -n 1 PlusPF.kraken_report.txt
</span><span id="__span-12-2"><a href="#__codelineno-12-2" id="__codelineno-12-2" name="__codelineno-12-2"></a>% of Seqs       Clades  Taxonomies      Rank    Taxonomy ID     Scientific Name
</span></code></pre></div>
The Zymo sample is prepared with 12% <em>Bacillus subtilis</em>.  Let's see how much
was actually reported in the results:
<div class="language-text highlight"><pre><span></span><code><span id="__span-13-1"><a href="#__codelineno-13-1" id="__codelineno-13-1" name="__codelineno-13-1"></a>(onyx) jovyan:~$ grep "Bacillus subtilis" PlusPF.kraken_report.txt
</span><span id="__span-13-2"><a href="#__codelineno-13-2" id="__codelineno-13-2" name="__codelineno-13-2"></a> 20.30  435278  1452    G1      653685                    Bacillus subtilis group
</span><span id="__span-13-3"><a href="#__codelineno-13-3" id="__codelineno-13-3" name="__codelineno-13-3"></a>  0.12  2624    1952    S       1423                        Bacillus subtilis
</span><span id="__span-13-4"><a href="#__codelineno-13-4" id="__codelineno-13-4" name="__codelineno-13-4"></a>  0.03  565     242     S1      135461                        Bacillus subtilis subsp. subtilis
</span><span id="__span-13-5"><a href="#__codelineno-13-5" id="__codelineno-13-5" name="__codelineno-13-5"></a>  0.01  108     108     S2      1404258                         Bacillus subtilis subsp. subtilis str. OH 131.1
</span><span id="__span-13-6"><a href="#__codelineno-13-6" id="__codelineno-13-6" name="__codelineno-13-6"></a>  ...
</span></code></pre></div>
Looks like 20.3%, though classified under <em>Bacillus subtilis</em> "subgroup",
rather than <em>Bacillus subtilis</em>, which reportedly only comprises 0.12% of the sample.
Most of that 20.3% is under <em>Bacillus spizizenii</em>.</p>
<p>An important detail here is that the fraction reported in this output
is not calculated in the same way as what's used in the reference values (12% for bacteria; 2% for yeasts).
Let's make a fairer comparison using the JSON taxonomic data.</p>
<h2 id="working-with-database-output-in-python">Working with database output in Python</h2>
<p>To fairly compare the taxonomic data with the reference values in the
Zymo community, we need to know the proportions of gDNA, so we need to
compute the number of base pairs that were assigned to each taxon.
Let's make this comparison in Python using the Onyx client's Python
API.</p>
<p>Let's first run the same query for the Zymo data.  We'll follow the
examples in the Onyx documentation and run the query in a context
manager.
<div class="language-py highlight"><pre><span></span><code><span id="__span-14-1"><a href="#__codelineno-14-1" id="__codelineno-14-1" name="__codelineno-14-1"></a><span class="kn">import</span> <span class="nn">os</span>
</span><span id="__span-14-2"><a href="#__codelineno-14-2" id="__codelineno-14-2" name="__codelineno-14-2"></a><span class="kn">from</span> <span class="nn">onyx</span> <span class="kn">import</span> <span class="n">OnyxConfig</span><span class="p">,</span> <span class="n">OnyxEnv</span><span class="p">,</span> <span class="n">OnyxClient</span>
</span><span id="__span-14-3"><a href="#__codelineno-14-3" id="__codelineno-14-3" name="__codelineno-14-3"></a>
</span><span id="__span-14-4"><a href="#__codelineno-14-4" id="__codelineno-14-4" name="__codelineno-14-4"></a><span class="n">config</span> <span class="o">=</span> <span class="n">OnyxConfig</span><span class="p">(</span>
</span><span id="__span-14-5"><a href="#__codelineno-14-5" id="__codelineno-14-5" name="__codelineno-14-5"></a>    <span class="n">domain</span><span class="o">=</span><span class="n">os</span><span class="o">.</span><span class="n">environ</span><span class="p">[</span><span class="n">OnyxEnv</span><span class="o">.</span><span class="n">DOMAIN</span><span class="p">],</span>
</span><span id="__span-14-6"><a href="#__codelineno-14-6" id="__codelineno-14-6" name="__codelineno-14-6"></a>    <span class="n">token</span><span class="o">=</span><span class="n">os</span><span class="o">.</span><span class="n">environ</span><span class="p">[</span><span class="n">OnyxEnv</span><span class="o">.</span><span class="n">TOKEN</span><span class="p">],</span>
</span><span id="__span-14-7"><a href="#__codelineno-14-7" id="__codelineno-14-7" name="__codelineno-14-7"></a><span class="p">)</span>
</span><span id="__span-14-8"><a href="#__codelineno-14-8" id="__codelineno-14-8" name="__codelineno-14-8"></a>
</span><span id="__span-14-9"><a href="#__codelineno-14-9" id="__codelineno-14-9" name="__codelineno-14-9"></a><span class="k">with</span> <span class="n">OnyxClient</span><span class="p">(</span><span class="n">config</span><span class="p">)</span> <span class="k">as</span> <span class="n">client</span><span class="p">:</span>
</span><span id="__span-14-10"><a href="#__codelineno-14-10" id="__codelineno-14-10" name="__codelineno-14-10"></a>    <span class="n">records</span> <span class="o">=</span> <span class="nb">list</span><span class="p">(</span><span class="n">client</span><span class="o">.</span><span class="n">filter</span><span class="p">(</span>
</span><span id="__span-14-11"><a href="#__codelineno-14-11" id="__codelineno-14-11" name="__codelineno-14-11"></a>        <span class="s2">"mscapetest"</span><span class="p">,</span>
</span><span id="__span-14-12"><a href="#__codelineno-14-12" id="__codelineno-14-12" name="__codelineno-14-12"></a>        <span class="n">fields</span><span class="o">=</span><span class="p">{</span>
</span><span id="__span-14-13"><a href="#__codelineno-14-13" id="__codelineno-14-13" name="__codelineno-14-13"></a>            <span class="s2">"extraction_enrichment_protocol__icontains"</span><span class="p">:</span> <span class="s2">"zymo"</span><span class="p">,</span>
</span><span id="__span-14-14"><a href="#__codelineno-14-14" id="__codelineno-14-14" name="__codelineno-14-14"></a>        <span class="p">},</span>
</span><span id="__span-14-15"><a href="#__codelineno-14-15" id="__codelineno-14-15" name="__codelineno-14-15"></a>    <span class="p">))</span>
</span></code></pre></div>
We've wrapped the <code>filter</code> call in a <code>list</code> because otherwise
we get a generator.</p>
<p>If you want to inspect the data, it's a bit easier to read if formatted with
indentation, which can be done using the standard <code>json.dumps</code> function:
<div class="language-py highlight"><pre><span></span><code><span id="__span-15-1"><a href="#__codelineno-15-1" id="__codelineno-15-1" name="__codelineno-15-1"></a><span class="kn">import</span> <span class="nn">json</span>
</span><span id="__span-15-2"><a href="#__codelineno-15-2" id="__codelineno-15-2" name="__codelineno-15-2"></a><span class="nb">print</span><span class="p">(</span><span class="n">json</span><span class="o">.</span><span class="n">dumps</span><span class="p">(</span><span class="n">records</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">indent</span><span class="o">=</span><span class="mi">2</span><span class="p">))</span>  <span class="c1"># show first record</span>
</span></code></pre></div>
In each record, the <code>'taxa'</code> key gives us a list of dictionaries
that each has a number of reads and a mean length, the product of
which is the total number of base pairs that were read for that
taxon.  A simple first step is to convert the taxonomic data (for the first record)
into a Pandas DataFrame with
<div class="language-py highlight"><pre><span></span><code><span id="__span-16-1"><a href="#__codelineno-16-1" id="__codelineno-16-1" name="__codelineno-16-1"></a><span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
</span><span id="__span-16-2"><a href="#__codelineno-16-2" id="__codelineno-16-2" name="__codelineno-16-2"></a>
</span><span id="__span-16-3"><a href="#__codelineno-16-3" id="__codelineno-16-3" name="__codelineno-16-3"></a><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">records</span><span class="p">[</span><span class="mi">0</span><span class="p">][</span><span class="s1">'taxa'</span><span class="p">])</span>
</span></code></pre></div>
We also need to drop a few lower-level taxa that are already
accounted for in higher ones. e.g. the reads for <em>Bacillus spizizenii TU-B-10</em> are
among the reads counted for <em>Bacillus spizizenii</em>.  A quick way of doing this
is by selecting the rows that have only two words in their names.
<div class="language-py highlight"><pre><span></span><code><span id="__span-17-1"><a href="#__codelineno-17-1" id="__codelineno-17-1" name="__codelineno-17-1"></a><span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="o">.</span><span class="n">loc</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="s1">'human_readable'</span><span class="p">]</span><span class="o">.</span><span class="n">apply</span><span class="p">(</span><span class="k">lambda</span> <span class="n">name</span><span class="p">:</span> <span class="nb">len</span><span class="p">(</span><span class="n">name</span><span class="o">.</span><span class="n">split</span><span class="p">())</span> <span class="o">==</span> <span class="mi">2</span><span class="p">)]</span>
</span></code></pre></div>
Now, let's add columns for the total number of base pairs associated with
each taxon and what proportion that is of the total.
<div class="language-py highlight"><pre><span></span><code><span id="__span-18-1"><a href="#__codelineno-18-1" id="__codelineno-18-1" name="__codelineno-18-1"></a><span class="n">df</span><span class="p">[</span><span class="s1">'gDNA'</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">'n_reads'</span><span class="p">]</span><span class="o">*</span><span class="n">df</span><span class="p">[</span><span class="s1">'mean_len'</span><span class="p">]</span>
</span><span id="__span-18-2"><a href="#__codelineno-18-2" id="__codelineno-18-2" name="__codelineno-18-2"></a><span class="n">df</span><span class="p">[</span><span class="s1">'proportion'</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="s1">'gDNA'</span><span class="p">]</span><span class="o">/</span><span class="n">df</span><span class="p">[</span><span class="s1">'gDNA'</span><span class="p">]</span><span class="o">.</span><span class="n">sum</span><span class="p">()</span>
</span></code></pre></div>
Finally, let's make a rough plot with a black dashed line at 12%.
<div class="language-py highlight"><pre><span></span><code><span id="__span-19-1"><a href="#__codelineno-19-1" id="__codelineno-19-1" name="__codelineno-19-1"></a><span class="kn">import</span> <span class="nn">matplotlib.pyplot</span> <span class="k">as</span> <span class="nn">plt</span>
</span><span id="__span-19-2"><a href="#__codelineno-19-2" id="__codelineno-19-2" name="__codelineno-19-2"></a>
</span><span id="__span-19-3"><a href="#__codelineno-19-3" id="__codelineno-19-3" name="__codelineno-19-3"></a><span class="n">plt</span><span class="o">.</span><span class="n">plot</span><span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="s1">'human_readable'</span><span class="p">],</span> <span class="n">df</span><span class="p">[</span><span class="s1">'proportion'</span><span class="p">]</span><span class="o">*</span><span class="mi">100</span><span class="p">,</span> <span class="s1">'o'</span><span class="p">)</span>
</span><span id="__span-19-4"><a href="#__codelineno-19-4" id="__codelineno-19-4" name="__codelineno-19-4"></a><span class="n">plt</span><span class="o">.</span><span class="n">axhline</span><span class="p">(</span><span class="mi">12</span><span class="p">,</span> <span class="n">c</span><span class="o">=</span><span class="s1">'k'</span><span class="p">,</span> <span class="n">ls</span><span class="o">=</span><span class="s1">'--'</span><span class="p">);</span>
</span><span id="__span-19-5"><a href="#__codelineno-19-5" id="__codelineno-19-5" name="__codelineno-19-5"></a><span class="n">plt</span><span class="o">.</span><span class="n">xticks</span><span class="p">(</span><span class="n">rotation</span><span class="o">=</span><span class="mf">22.5</span><span class="p">,</span> <span class="n">ha</span><span class="o">=</span><span class="s1">'right'</span><span class="p">);</span>
</span></code></pre></div></p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-type="image" data-width="100%" href="./zymo-comparison.png"><img alt="Measured gDNA proportions of a Zymo community" src="./zymo-comparison.png"/></a></p>
<p>There are some clear discrepancies—<em>Pseudomonas aeruginosa</em> is
underreported and <em>Bacillus spizizenii</em> is overreported—but this
matches results by e.g. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6520541/">Nicholls et
al. (2019)</a>.</p>
<p>This short example is intended as a basic demonstration of what's
possible in CLIMB-TRE.  We're always interested to hear more examples
of research questions that CLIMB-TRE can answer, so let us know if you
have an example that could be included as a guide for others.</p>
</div>
</div><footer>
<hr/>
<div role="contentinfo">
<!-- Copyright etc -->
</div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
</div>
</div>
</section>
</div>
<div aria-label="Versions" class="rst-versions" role="note">
<span class="rst-current-version" data-toggle="rst-current-version">
<span>
<a class="fa fa-github" href="https://github.com/GSTT-CIDR/network_hub" style="color: #fcfcfc"> GitHub</a>
</span>
</span>
</div>
<script src="../js/jquery-3.6.0.min.js"></script>
<script>var base_url = "..";</script>
<script src="../js/theme_extra.js"></script>
<script src="../js/theme.js"></script>
<script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>
<script id="init-glightbox">const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});
</script></body>
</html>
