<!DOCTYPE html>

<html class="writer-html5" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="IE=edge" http-equiv="X-UA-Compatible"/>
<meta content="width=device-width, initial-scale=1.0" name="viewport"/><meta content="CIDR" name="author"/><link href="https://CIDR-network-hub.github.io/running_organism_query/" rel="canonical"/>
<link href="../img/favicon.ico" rel="shortcut icon"/>
<title>Query a classification - CIDR Metagenomics Hub</title>
<link href="../css/theme.css" rel="stylesheet"/>
<link href="../css/theme_extra.css" rel="stylesheet"/>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" rel="stylesheet"/>
<link href="../assets/_mkdocstrings.css" rel="stylesheet"/>
<link href="../assets/js/jquery-jvectormap.css" rel="stylesheet"/>
<link href="../assets/stylesheets/width.css" rel="stylesheet"/>
<link href="https://cdn.datatables.net/2.2.2/css/dataTables.dataTables.min.css" rel="stylesheet"/>
<script>
        // Current page data
        var mkdocs_page_name = "Query a classification";
        var mkdocs_page_input_path = "running_organism_query.md";
        var mkdocs_page_url = "/running_organism_query/";
      </script>
<!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
<script>hljs.highlightAll();</script>
<link href="../cidr_metagenomics.pdf" rel="alternate" title="PDF" type="application/pdf"/> <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }</style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
<body class="wy-body-for-nav" role="document">
<div class="wy-grid-for-nav">
<nav class="wy-nav-side stickynav" data-toggle="wy-nav-shift">
<div class="wy-side-scroll">
<div class="wy-side-nav-search">
<a class="icon icon-home" href=".."> CIDR Metagenomics Hub
        </a>
</div>
<div aria-label="Navigation menu" class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation">
<p class="caption"><span class="caption-text">Introduction</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="..">Clinical metagenomics</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Lab Resources</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../panmetagenomics_protocol/">Lab protocols</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Bioinformatics</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_outline/">Overview</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_install/">Installation and validation</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="#">Launching a sequencing experiment using ONT software</a>
<ul>
<li class="toctree-l2"><a class="reference internal" href="../running_minknow/">MinKNOW</a>
</li>
<li class="toctree-l2"><a class="reference internal" href="../running_gourami/">Gourami</a>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../running_metagenomics_workflow/">Running the metagenomics workflow</a>
</li>
<li class="toctree-l1 current"><a class="reference internal current" href="./">Query a classification</a>
<ul class="current">
<li class="toctree-l2"><a class="reference internal" href="#technical-information">Technical information</a>
</li>
<li class="toctree-l2"><a class="reference internal" href="#launching-organism-query">Launching organism query</a>
</li>
<li class="toctree-l2"><a class="reference internal" href="#interpreting-results">Interpreting results</a>
<ul>
<li class="toctree-l3"><a class="reference internal" href="#interactive-plot">Interactive plot</a>
</li>
<li class="toctree-l3"><a class="reference internal" href="#blast-alignments">BLAST alignments</a>
</li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../mSCAPE_integration/">Uploading to mSCAPE</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../additional_tools/">Summary report</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../alternative_deployment/">Deployment alternatives</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../clinical_interpretation/">Clinical evaluation SOP</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../validation_outline/">Validation outline</a>
</li>
</ul>
<p class="caption"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq/">Frequently asked questions</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../genome_database/">Metagenomics sequence database</a>
</li>
</ul>
</div>
</div>
</nav>
<section class="wy-nav-content-wrap" data-toggle="wy-nav-shift">
<nav aria-label="Mobile navigation menu" class="wy-nav-top" role="navigation">
<i class="fa fa-bars" data-toggle="wy-nav-top"></i>
<a href="..">CIDR Metagenomics Hub</a>
</nav>
<div class="wy-nav-content">
<div class="rst-content"><div aria-label="breadcrumbs navigation" role="navigation">
<ul class="wy-breadcrumbs">
<li><a aria-label="Docs" class="icon icon-home" href=".."></a></li>
<li class="breadcrumb-item">Bioinformatics</li>
<li class="breadcrumb-item active">Query a classification</li>
<li class="wy-breadcrumbs-aside">
<a class="icon icon-github" href="https://github.com/GSTT-CIDR/network_hub/edit/master/docs/running_organism_query.md"> Edit on GitHub</a>
</li>
</ul>
<hr/>
</div>
<div class="document" itemscope="itemscope" itemtype="http://schema.org/Article" role="main">
<div class="section" itemprop="articleBody">
<div><h1 id="bioinformatics-organism-query">Bioinformatics - Organism query</h1>
<p>The Organism Query tool is designed to help scrutinise taxonomic classification outputs from the CIDR metagenomics workflow. It uses a local (offline) version of NCBI BLASTn, with the full NCBI nt, RefSeq, and CIDR databases to produce a report, similar to that found on the NCBI BLAST website, providing the user with a second opinion on classifications.</p>
<h2 id="technical-information">Technical information</h2>
<ul>
<li>The tool searches the classified reads for an organism indicated by the user. Selecting a random subset of 50 (max) reads assigned to that taxa.</li>
<li>The centrifuge score is not used in this analysis, all reads matching a taxon are valid for selection</li>
<li>Reads are extracted from the microbial FASTQ file stored in the workflow <code>results</code> folder and BLASTed against the prescribed database. The subsetted reads are saved as a FASTA file in the <code>./metagenomics/reports/{sample_id}/organism_query_XXX</code> directory.</li>
<li>The results are parsed in to a HTML report with an interactive plot designed to help the user explore the different metrics of alignment.</li>
<li>A full BLAST alignment report is available at the bottom of the HTML report.</li>
<li>The report is stored in the <code>./metagenomics/reports/{sample_id}/organism_query_XXX</code> folder, with the other PDF reports from the metagenomics run.</li>
</ul>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-type="image" data-width="auto" href="../img/organism_query_blank.png"><img alt="alt text" src="../img/organism_query_blank.png"/></a></p>
<h2 id="launching-organism-query">Launching organism query</h2>
<ol>
<li>Load the relevant report for the run you'd like to query.
<br/><br/></li>
<li>Click on the Organism Query launcher desktop icon.
<br/><br/></li>
<li>Fill out the fields detailed below. Multiple queries can be run at once by selecting the <code>+</code> button. A separate report will be generated for each.</li>
</ol>
<table>
<thead>
<tr>
<th>Parameter</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>CIDR workflow Lab/sample ID</strong></td>
<td>The Lab/Sample ID matching that of the report in question.</td>
</tr>
<tr>
<td><strong>Workflow hour/interval</strong></td>
<td>The time-point corresponding to the dataset you'd like query.</td>
</tr>
<tr>
<td><strong>Organism keyword</strong></td>
<td>A keyword identifying the taxa to be queried eg. 'Aspergillus' (capturing all aspergillus spp.) or 'Bordetella parapertussis' for this species and all taxonomic children (eg strains)</td>
</tr>
</tbody>
</table>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-type="image" data-width="auto" href="../videos/organism_query.mp4"><div class="video-container"><video alt="type:video" controls="" style="width:50%;height:35%"><source src="../videos/organism_query.mp4" type="video/mp4"/></video></div></a></p>
<ol start="4">
<li>Click on the <code>Launch script</code> button to start the query workflow. A Firefox browser window will appear after the workflow has finished. You can reopen the report from <code>`reports/{sample_id}/organism_query_XXX</code> on the metagenomics SSD. </li>
</ol>
<h2 id="interpreting-results">Interpreting results</h2>
<p>Interpreting the results can be subjective. The interactive plot has been designed to help guide decision making. Using a combination of the alignment length (relative to the read length), the identity score and the e-score can be an informative approach.</p>
<ul>
<li>Query Coverage: Percent of the query sequence length that is included in alignments against the sequence match.</li>
<li>E-value: Indicates the number of hits or alignments that are expected to be seen by random chance with the same score or better. The lower the E-value, the more significant the alignment (the closer to 0, the better). E-value is the default metric used to sort the Descriptions table. Click here for a discussion of E-value thresholds.</li>
<li>Percent Identity: Percent of nucleotides or amino acids that are identical between the aligned query and database sequences. A query sequence can share low percent identity with a sequence and still be a significant hit. It is essential to take the E-value into account and look for similarity between conserved regions (this will be more evident at the amino acid level).</li>
</ul>
<h3 id="interactive-plot">Interactive plot</h3>
<p>In a situation where there is either no confident match or no convergence across the read subset on an organism, or a alignments indicating a number of similar, closely related organisms, the plot appears disordered in a single cluster.
<a class="glightbox" data-desc-position="bottom" data-height="auto" data-type="image" data-width="auto" href="../img/organism_query_plot_undecided.png"><img alt="alt text" src="../img/organism_query_plot_undecided.png"/></a></p>
<p>Situations where a convergent set of alignments are indicated results in a plot with two clusters, usually with one clear homogenous taxa. In this case, the Tobacco Mosaic virus.</p>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-type="image" data-width="auto" href="../img/plot_decided.png"><img alt="alt text" src="../img/plot_decided.png"/></a></p>
<h3 id="blast-alignments">BLAST alignments</h3>
<p>Scrolling to the bottom of the report is a conventional BLAST alignment. We recommend looking at each read, the <strong>Length</strong> (query readlength), the <strong>Score</strong> and <strong>Identities</strong> values. </p>
<ul>
<li>If only a small alignment (80 bp) has been made from a long 2000 bp read, this is not likely to be a robust estimation. </li>
<li>Check the alignment visualisation for long repeats, regions of low complexity etc. These regions often confound BLAST and taxonomic classification tools.</li>
</ul>
<p><a class="glightbox" data-desc-position="bottom" data-description="An example of a Organism Query BLAST output" data-height="auto" data-title="BLAST screenshot" data-type="image" data-width="auto" href="../img/organism_query_blast.png"><img alt="" data-description="An example of a Organism Query BLAST output" data-title="BLAST screenshot" src="../img/organism_query_blast.png"/></a></p></div>
</div>
</div><footer>
<div aria-label="Footer Navigation" class="rst-footer-buttons" role="navigation">
<a class="btn btn-neutral float-left" href="../running_metagenomics_workflow/" title="Running the metagenomics workflow"><span class="icon icon-circle-arrow-left"></span> Previous</a>
<a class="btn btn-neutral float-right" href="../mSCAPE_integration/" title="Uploading to mSCAPE">Next <span class="icon icon-circle-arrow-right"></span></a>
</div>
<hr/>
<div role="contentinfo">
<!-- Copyright etc -->
</div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
</div>
</div>
</section>
</div>
<div aria-label="Versions" class="rst-versions" role="note">
<span class="rst-current-version" data-toggle="rst-current-version">
<span>
<a class="fa fa-github" href="https://github.com/GSTT-CIDR/network_hub" style="color: #fcfcfc"> GitHub</a>
</span>
<span><a href="../running_metagenomics_workflow/" style="color: #fcfcfc">« Previous</a></span>
<span><a href="../mSCAPE_integration/" style="color: #fcfcfc">Next »</a></span>
</span>
</div>
<script src="../js/jquery-3.6.0.min.js"></script>
<script>var base_url = "..";</script>
<script src="../js/theme_extra.js"></script>
<script src="../js/theme.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
<script src="https://cdn.datatables.net/2.2.2/js/dataTables.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jvectormap/2.0.5/jquery-jvectormap.js"></script>
<script src="../assets/js/jquery-jvectormap.js"></script>
<script src="../assets/js/jquery-mousewheel.js"></script>
<script src="../assets/js/jvectormap.js"></script>
<script src="../assets/js/abstract-element.js"></script>
<script src="../assets/js/abstract-canvas-element.js"></script>
<script src="../assets/js/abstract-shape-element.js"></script>
<script src="../assets/js/svg-element.js"></script>
<script src="../assets/js/svg-group-element.js"></script>
<script src="../assets/js/svg-canvas-element.js"></script>
<script src="../assets/js/svg-shape-element.js"></script>
<script src="../assets/js/svg-path-element.js"></script>
<script src="../assets/js/svg-circle-element.js"></script>
<script src="../assets/js/svg-image-element.js"></script>
<script src="../assets/js/svg-text-element.js"></script>
<script src="../assets/js/vml-element.js"></script>
<script src="../assets/js/vml-group-element.js"></script>
<script src="../assets/js/vml-canvas-element.js"></script>
<script src="../assets/js/vml-shape-element.js"></script>
<script src="../assets/js/vml-path-element.js"></script>
<script src="../assets/js/vml-circle-element.js"></script>
<script src="../assets/js/vml-image-element.js"></script>
<script src="../assets/js/map-object.js"></script>
<script src="../assets/js/region.js"></script>
<script src="../assets/js/marker.js"></script>
<script src="../assets/js/vector-canvas.js"></script>
<script src="../assets/js/simple-scale.js"></script>
<script src="../assets/js/ordinal-scale.js"></script>
<script src="../assets/js/numeric-scale.js"></script>
<script src="../assets/js/color-scale.js"></script>
<script src="../assets/js/legend.js"></script>
<script src="../assets/js/data-series.js"></script>
<script src="../assets/js/proj.js"></script>
<script src="../assets/js/map.js"></script>
<script src="../assets/js/uk_mill.js"></script>
<script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>
<script>const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});</script></body>
</html>
