<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="CIDR" /><link rel="canonical" href="https://CIDR-network-hub.github.io/stable/alternative_deployment/" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>Deployment alternatives - CIDR Metagenomics Hub</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
        <link href="../assets/_mkdocstrings.css" rel="stylesheet" />
        <link href="../assets/js/jquery-jvectormap.css" rel="stylesheet" />
        <link href="https://cdn.datatables.net/2.2.2/css/dataTables.dataTables.min.css" rel="stylesheet" />
        <link href="../css/version-select.css" rel="stylesheet" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Deployment alternatives";
        var mkdocs_page_input_path = "alternative_deployment.md";
        var mkdocs_page_url = "/stable/alternative_deployment/";
      </script>
    
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
      <script>hljs.highlightAll();</script> 
 <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }</style> <script src="../assets/javascripts/glightbox.min.js"></script></head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> CIDR Metagenomics Hub
        </a>
        <div class="version">
          {'provider': 'mike', 'default': 'latest'}
        </div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Introduction</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="..">Clinical metagenomics</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Lab Resources</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../panmetagenomics_protocol/">Lab protocols</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Bioinformatics</span></p>
              <ul class="current">
                  <li class="toctree-l1"><a class="reference internal" href="../bioinformatics_outline/">Overview</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../bioinformatics_install/">Installation and validation</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../running_minknow/">Starting a MinKNOW experiment</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../running_metagenomics_workflow/">Running the metagenomics workflow</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../running_organism_query/">Query a classification</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../mSCAPE_integration/">Uploading to mSCAPE</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../additional_tools/">Summary report</a>
                  </li>
                  <li class="toctree-l1 current"><a class="reference internal current" href="./">Deployment alternatives</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#overview">Overview</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#install-instructions">Install instructions</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#install-validation">Install validation</a>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#implementation">Implementation</a>
    </li>
    </ul>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Analysis</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../clinical_interpretation/">Clinical evaluation SOP</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../validation_outline/">Validation outline</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">FAQ</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../faq/">Frequently asked questions</a>
                  </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">CIDR Metagenomics Hub</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" aria-label="Docs"></a></li>
          <li class="breadcrumb-item">Bioinformatics</li>
      <li class="breadcrumb-item active">Deployment alternatives</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/GSTT-CIDR/network_hub/edit/master/docs/alternative_deployment.md" class="icon icon-github"> Edit on GitHub</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <div><h1 id="setting-up-cidr-metagenomics-bioinformatics-workflow-alternative-deployment">Setting up CIDR Metagenomics bioinformatics workflow - alternative deployment</h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p><strong>The tools and documentation described here and on the CIDR GitHub are not validated for diagnostic use and are for research and evaluation purposes only.</strong></p>
</div>
<h2 id="overview">Overview</h2>
<p>For collaborators outside of the Network, an alternative configuration can be provided. This will bypass the GUI allowing users to provide a <code>sample_sheet.csv</code> through a CLI. Organism query will not be available to headless users as this tools is heavily reliant on GUI I/O.</p>
<h2 id="install-instructions">Install instructions</h2>
<ol>
<li>Decompress CIDR_metagenomics_vX.X.tar.gz: </li>
</ol>
<pre><code>tar -xvzf CIDR_metagenomics_vX.X.tar.gz
</code></pre>
<ol start="2">
<li>Install conda/<a href="https://mamba.readthedocs.io/en/latest/installation/mamba-installation.html">mamba</a>.</li>
<li>Build the appropriate environment for running the CIDR metagenomics containers. </li>
</ol>
<pre><code>wget https://raw.githubusercontent.com/GSTT-CIDR/metagenomics_container/main/conda/apptainer.yml
conda env create -f apptainer.yml

</code></pre>
<ol start="4">
<li>Allocate a directory for MinKNOW data outputs. This will be mounted to the <code>/data</code> directory in the container in a later step.</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The directory structure of data for ingest must be maintained as in standard MinKNOW outputs eg.
<strong>Example for control sample</strong>
<code>[minknow_outputs_directory]/GSTT_control_sample_01/GSTT_control_sample_01/20240424_1408_X4_FAY88387_d3868a4f/fastq_pass/barcode11</code>
<strong>Naming schema</strong>
<code>[minknow_outputs_directory]/[experiemnt]/[sample_id]/[*]/fastq_pass/barcodeXX</code></p>
</div>
<h2 id="install-validation">Install validation</h2>
<ol start="5">
<li>Navigate to the root of the <code>CIDR_metagenomics_vX.X</code> directory.</li>
<li>Move <code>CIDR_metagenomics_vX.X/GSTT_control_sample_XX</code> to the allocated directory for MinKNOW data outputs (from Install instructions: Step 4).</li>
<li>activate the apptainer conda environment:
<code>conda activate apptainer</code></li>
<li>Initiate the run for analysing the control dataset:</li>
</ol>
<pre><code>apptainer exec --bind ./:/mnt  --bind ./data:/data  ./containers/cidr_metagenomics_v3.6.sif bash -c 'cd /workflow ; source /opt/conda/etc/profile.d/conda.sh &amp;&amp; conda activate cmg &amp;&amp; for t in 0.5; do snakemake --directory /mnt --cores 20 -k --config time=$t samples=/mnt/sample_sheets/CIDR_control_1.csv --latency-wait 15; done'

</code></pre>
<ol start="8">
<li>When the workflow has completed, inspect the <code>CIDR_metagenomics_vX.X/reports/CIDR_control_1</code> PDF report, it should match the CIDR validation report provided <a href="./CIDR_control_1_0.5_hours_report.pdf">here</a>.</li>
</ol>
<div class="admonition info">
<p class="admonition-title">Info</p>
<p>Variables to change in step 3<br>
<strong>--bind ./:/mnt</strong> - Binding the workflow root directory to the container /mnt.<br>
<strong>--bind ./data:/data</strong> - binding the allocated directory for MinKNOW data outputs to /data.<br>
<strong>./containers/cidr_metagenomics_v3.6.sif</strong> -  launching the metagenomics container.<br>
<strong>for t in 0.5 1 2 16 24</strong> - time-points for analysis.<br>
<strong>--cores 20</strong> - number of samples to be processed simultaneously - not the same as threads.<br>
<strong>samples=/mnt/sample_sheets/CIDR_control_1.csv</strong> - the mounted path for the sample sheet - remember this is the relative mounted path, so <code>/mnt/sample_sheets</code> corresponds to <code>CIDR_metagenomics_vX.X/sample_sheets</code> on the host machine.<br></p>
</div>
<h2 id="implementation">Implementation</h2>
<ol start="9">
<li>Build a <strong>sample sheet</strong> copying the structure of the example in <code>CIDR_metagenomics_vX.X/sample_sheets</code>. Importantly, 'Experiment', 'SampleID' and 'Barcode' must be correct and correspond to the <code>[minknow_outputs_directory]/[experiemnt]/[sample_id]/[*]/fastq_pass/barcodeXX</code> scheme.</li>
<li>activate the apptainer conda environment:
<code>conda activate apptainer</code></li>
<li>Run the container, changing the flags explained in the validation step:</li>
</ol>
<pre><code>apptainer exec --bind ./:/mnt  --bind ./data:/data  ./containers/cidr_metagenomics_v3.6.sif bash -c 'cd /workflow ; source /opt/conda/etc/profile.d/conda.sh &amp;&amp; conda activate cmg &amp;&amp; for t in 0.5 1 2 16 24; do snakemake --directory /mnt --cores 20 -k --config time=$t samples=/mnt/sample_sheets/[**sample_sheet**] --latency-wait 15; done'

</code></pre>
<ol start="11">
<li>PDF outputs should be found in <code>CIDR_metagenomics_vX.X/reports/</code> corresponding to each LabID in the <strong>sample sheet</strong> loaded.</li>
</ol></div>
              
            </div>
          </div><footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="Footer Navigation">
        <a href="../additional_tools/" class="btn btn-neutral float-left" title="Summary report"><span class="icon icon-circle-arrow-left"></span> Previous</a>
        <a href="../clinical_interpretation/" class="btn btn-neutral float-right" title="Clinical evaluation SOP">Next <span class="icon icon-circle-arrow-right"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/GSTT-CIDR/network_hub" class="fa fa-github" style="color: #fcfcfc"> GitHub</a>
        </span>
    
    
      <span><a href="../additional_tools/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../clinical_interpretation/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script src="../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "..";</script>
    <script src="../js/theme_extra.js"></script>
    <script src="../js/theme.js"></script>
      <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
      <script src="https://cdn.datatables.net/2.2.2/js/dataTables.min.js"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jvectormap/2.0.5/jquery-jvectormap.js"></script>
      <script src="../assets/js/jquery-jvectormap.js"></script>
      <script src="../assets/js/jquery-mousewheel.js"></script>
      <script src="../assets/js/jvectormap.js"></script>
      <script src="../assets/js/abstract-element.js"></script>
      <script src="../assets/js/abstract-canvas-element.js"></script>
      <script src="../assets/js/abstract-shape-element.js"></script>
      <script src="../assets/js/svg-element.js"></script>
      <script src="../assets/js/svg-group-element.js"></script>
      <script src="../assets/js/svg-canvas-element.js"></script>
      <script src="../assets/js/svg-shape-element.js"></script>
      <script src="../assets/js/svg-path-element.js"></script>
      <script src="../assets/js/svg-circle-element.js"></script>
      <script src="../assets/js/svg-image-element.js"></script>
      <script src="../assets/js/svg-text-element.js"></script>
      <script src="../assets/js/vml-element.js"></script>
      <script src="../assets/js/vml-group-element.js"></script>
      <script src="../assets/js/vml-canvas-element.js"></script>
      <script src="../assets/js/vml-shape-element.js"></script>
      <script src="../assets/js/vml-path-element.js"></script>
      <script src="../assets/js/vml-circle-element.js"></script>
      <script src="../assets/js/vml-image-element.js"></script>
      <script src="../assets/js/map-object.js"></script>
      <script src="../assets/js/region.js"></script>
      <script src="../assets/js/marker.js"></script>
      <script src="../assets/js/vector-canvas.js"></script>
      <script src="../assets/js/simple-scale.js"></script>
      <script src="../assets/js/ordinal-scale.js"></script>
      <script src="../assets/js/numeric-scale.js"></script>
      <script src="../assets/js/color-scale.js"></script>
      <script src="../assets/js/legend.js"></script>
      <script src="../assets/js/data-series.js"></script>
      <script src="../assets/js/proj.js"></script>
      <script src="../assets/js/map.js"></script>
      <script src="../assets/js/uk_mill.js"></script>
      <script src="../js/version-select.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

<script>const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});</script></body>
</html>
