<!DOCTYPE html>

<html class="writer-html5" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="IE=edge" http-equiv="X-UA-Compatible"/>
<meta content="width=device-width, initial-scale=1.0" name="viewport"/><meta content="CIDR" name="author"/><link href="https://CIDR-network-hub.github.io/lab_landing/" rel="canonical"/>
<link href="../img/favicon.ico" rel="shortcut icon"/>
<title>Analysing data - CIDR Metagenomics Hub</title>
<link href="../css/theme.css" rel="stylesheet"/>
<link href="../css/theme_extra.css" rel="stylesheet"/>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" rel="stylesheet"/>
<link href="../assets/_mkdocstrings.css" rel="stylesheet"/>
<link href="../assets/js/jquery-jvectormap.css" rel="stylesheet"/>
<script>
        // Current page data
        var mkdocs_page_name = "Analysing data";
        var mkdocs_page_input_path = "lab_landing.md";
        var mkdocs_page_url = "/lab_landing/";
      </script>
<!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
<script>hljs.highlightAll();</script>
<link href="../cidr_metagenomics.pdf" rel="alternate" title="PDF" type="application/pdf"/> <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }</style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
<body class="wy-body-for-nav" role="document">
<div class="wy-grid-for-nav">
<nav class="wy-nav-side stickynav" data-toggle="wy-nav-shift">
<div class="wy-side-scroll">
<div class="wy-side-nav-search">
<a class="icon icon-home" href=".."> CIDR Metagenomics Hub
        </a>
</div>
<div aria-label="Navigation menu" class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation">
<ul>
<li class="toctree-l1"><a class="reference internal" href="..">Introduction</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Lab Resources</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../panmetagenomics_protocol/">Lab protocols</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Bioinformatics</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_outline/">Outline</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_install/">Installation</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../running_metagenomics_workflow/">Starting the metagenomics workflow</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../running_organism_query/">Query a classification</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../alternative_deployment/">Deployment alternatives</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../clinical_interpretation/">Clinical evaluation SOP</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../validation_outline/">Validation outline</a>
</li>
</ul>
</div>
</div>
</nav>
<section class="wy-nav-content-wrap" data-toggle="wy-nav-shift">
<nav aria-label="Mobile navigation menu" class="wy-nav-top" role="navigation">
<i class="fa fa-bars" data-toggle="wy-nav-top"></i>
<a href="..">CIDR Metagenomics Hub</a>
</nav>
<div class="wy-nav-content">
<div class="rst-content"><div aria-label="breadcrumbs navigation" role="navigation">
<ul class="wy-breadcrumbs">
<li><a aria-label="Docs" class="icon icon-home" href=".."></a></li>
<li class="breadcrumb-item active">Analysing data</li>
<li class="wy-breadcrumbs-aside">
<a class="icon icon-github" href="https://github.com/GSTT-CIDR/network_hub/edit/master/docs/lab_landing.md"> Edit on GitHub</a>
</li>
</ul>
<hr/>
</div>
<div class="document" itemscope="itemscope" itemtype="http://schema.org/Article" role="main">
<div class="section" itemprop="articleBody">
<div><h1 id="analysing-data">Analysing data</h1>
<h2 id="overview">Overview</h2>
<p>Once data and metadata have been ingested into the Onyx database, you
can query it using the Onyx client, which provides a command line interface (CLI)
and Python API.  This short example
demonstrates a few principal functions.  More are described in the
<a href="https://climb-tre.github.io/onyx-client/"><code>onyx-client</code> documentation</a>.</p>
<p>This guide also assumes that you're using a Notebook Server on CLIMB,
so that once installed, the Onyx client will automatically be configured.</p>
<h2 id="onyx-client-basics">Onyx client basics</h2>
<p>First, let's install the Onyx client, which is available through the
<a href="https://pypi.org/project/climb-onyx-client">Python package</a>
<code>climb-onyx-client</code> and can thus be installed
with <code>pip</code>.  As advised in the <a href="https://docs.climb.ac.uk/notebook-servers/installing-software-with-conda/">CLIMB docs on installing
software</a>,
you should install the client in a new Conda environment.
I'll name my environment <code>onyx</code> and install <code>ipykernel</code> too
so that it's available in my Jupyter Notebooks.</p>
<pre><code>jovyan:~$ conda create -n onyx ipykernel
</code></pre>
<p>Let's activate this environment and install the Onyx client.</p>
<pre><code>jovyan:~$ conda activate onyx
(onyx) jovyan:~$ pip install climb-onyx-client
</code></pre>
<p>On Bryn's Notebook Servers, the client will automatically be configured.
Try running the command-line client with</p>
<pre><code>(onyx) jovyan:~$ onyx
</code></pre>
<p>This should show you some options and commands that are available.
Have a look at your own profile with</p>
<pre><code>(onyx) jovyan:~$ onyx profile
</code></pre>
<p>and which projects you have access to with</p>
<pre><code>(onyx) jovyan:~$ onyx projects
</code></pre>
<p>You should see <code>mscapetest</code> listed.</p>
<h2 id="querying-data">Querying data</h2>
<p>As an example task, we'll see if we can find any sequencing data performed
for ZymoBIOMICS sources.  These are designed with 
<a href="https://files.zymoresearch.com/protocols/_d6300_zymobiomics_microbial_community_standard.pdf">a particular specification</a>
of DNA from eight bacteria and two yeasts.  We can use these to see if our protocol
correctly recovers the DNA fractions. I.e. if our protocol is biased.</p>
<p>From the command line, the main route to querying Onyx is via the <code>filter</code> command.
On its own, this queries the database with <em>no</em> filters.  The command</p>
<pre><code>(onyx) jovyan:~$ onyx filter mscapetest
</code></pre>
<p>will produce tens of thousands of lines of JSON, so let's not
do that just yet.  To first see which fields are available in the database,
we can use</p>
<pre><code>(onyx) jovyan:~$ onyx fields mscapetest
...
├────────────────────────────────┼──────────┼───────────────────┼──────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────┤
│ extraction_enrichment_protocol │ optional │ text              │ Details of nucleic acid extraction and optional enrichment steps.            │                                                                             │
├────────────────────────────────┼──────────┼───────────────────┼──────────────────────────────────────────────────────────────────────────────┼─────────────────────────────────────────────────────────────────────────────┤
...
</code></pre>
<p>Let's search the database for entries with <code>zymo</code> (case-insensitive) in this field.</p>
<pre><code>(onyx) jovyan:~$ onyx filter mscapetest --field extraction_enrichment_protocol.icontains=zymo
...
</code></pre>
<p>That should return JSON data for a few entries.  You may wish to format the
data as CSV or TSV with <code>--format csv</code> or <code>--format tsv</code>, respectively.</p>
<h2 id="inspecting-some-pipeline-output-on-the-command-line">Inspecting some pipeline output on the command line</h2>
<p>When data is ingested into Onyx, a taxonomic classification is automatically run.
The last part of the JSON data is usually some of this, in JSON format.
The complete reports can be found in the S3 buckets given in the
<code>'taxon_report'</code> field.  You can find this in the output you've already produced
or modify the <code>filter</code> command to only request them using the <code>--include</code> flag. e.g.</p>
<pre><code>(onyx) jovyan:~$ onyx filter mscapetest --field extraction_enrichment_protocol.icontains=zymo --include=taxon_reports
[
    {
        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-FDE50853AD/"
    },
    {
        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-04F4495068/"
    }
]
</code></pre>
<p>If you'd like multiple fields, enter each one with another <code>--include</code> flag,
or use bash's <a href="https://www.gnu.org/software/bash/manual/html_node/Brace-Expansion.html">brace expansion</a>, e.g.</p>
<pre><code>(onyx) jovyan:~$ onyx filter mscapetest --field extraction_enrichment_protocol.icontains=zymo --include={cid,taxon_reports}
[
    {
        "cid": "C-FDE50853AD",
        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-FDE50853AD/"
    },
    {
        "cid": "C-04F4495068",
        "taxon_reports": "s3://mscapetest-published-taxon-reports/C-04F4495068/"
    }
]
</code></pre>
<p>You can conversely exclude individual fields using the <code>--exclude</code>
flag in the same way.</p>
<p>Either way, you now have the location of the taxonomy reports.  Let's have a look
with <code>s3cmd</code>.</p>
<pre><code>(onyx) jovyan:~$ s3cmd ls s3://mscapetest-published-taxon-reports/C-FDE50853AD/
2023-11-10 12:56   146K  s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken.json
2023-11-10 12:56     2G  s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_assignments.tsv
2023-11-10 12:56   193K  s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_report.txt
</code></pre>
<p>The plain text report is what we're after, so let's download that with <code>s3cmd</code>:</p>
<pre><code>(onyx) jovyan:~$ s3cmd get s3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_report.txt
download: 's3://mscapetest-published-taxon-reports/C-FDE50853AD/PlusPF.kraken_report.txt' -&gt; './PlusPF.kraken_report.txt'  [1 of 1]
 197750 of 197750   100% in    0s     3.79 MB/s  done
</code></pre>
<p>If you've never seen one of these reports before, it's worth having a
quick look with a tool like <code>less</code> or by opening it using the
JupyterLab file browser.  For reference, it's worth showing the header</p>
<pre><code>(onyx) jovyan:~$ head -n 1 PlusPF.kraken_report.txt
% of Seqs       Clades  Taxonomies      Rank    Taxonomy ID     Scientific Name
</code></pre>
<p>The Zymo sample is prepared with 12% <em>Bacillus subtilis</em>.  Let's see how much
was actually reported in the results:</p>
<pre><code>(onyx) jovyan:~$ grep "Bacillus subtilis" PlusPF.kraken_report.txt
 20.30  435278  1452    G1      653685                    Bacillus subtilis group
  0.12  2624    1952    S       1423                        Bacillus subtilis
  0.03  565     242     S1      135461                        Bacillus subtilis subsp. subtilis
  0.01  108     108     S2      1404258                         Bacillus subtilis subsp. subtilis str. OH 131.1
  ...
</code></pre>
<p>Looks like 20.3%, though classified under <em>Bacillus subtilis</em> "subgroup",
rather than <em>Bacillus subtilis</em>, which reportedly only comprises 0.12% of the sample.
Most of that 20.3% is under <em>Bacillus spizizenii</em>.</p>
<p>An important detail here is that the fraction reported in this output
is not calculated in the same way as what's used in the reference values (12% for bacteria; 2% for yeasts).
Let's make a fairer comparison using the JSON taxonomic data.</p>
<h2 id="working-with-database-output-in-python">Working with database output in Python</h2>
<p>To fairly compare the taxonomic data with the reference values in the
Zymo community, we need to know the proportions of gDNA, so we need to
compute the number of base pairs that were assigned to each taxon.
Let's make this comparison in Python using the Onyx client's Python
API.</p>
<p>Let's first run the same query for the Zymo data.  We'll follow the
examples in the Onyx documentation and run the query in a context
manager.</p>
<pre><code class="language-py">import os
from onyx import OnyxConfig, OnyxEnv, OnyxClient

config = OnyxConfig(
    domain=os.environ[OnyxEnv.DOMAIN],
    token=os.environ[OnyxEnv.TOKEN],
)

with OnyxClient(config) as client:
    records = list(client.filter(
        "mscapetest",
        fields={
            "extraction_enrichment_protocol__icontains": "zymo",
        },
    ))
</code></pre>
<p>We've wrapped the <code>filter</code> call in a <code>list</code> because otherwise
we get a generator.</p>
<p>If you want to inspect the data, it's a bit easier to read if formatted with
indentation, which can be done using the standard <code>json.dumps</code> function:</p>
<pre><code class="language-py">import json
print(json.dumps(records[0], indent=2))  # show first record
</code></pre>
<p>In each record, the <code>'taxa'</code> key gives us a list of dictionaries
that each has a number of reads and a mean length, the product of
which is the total number of base pairs that were read for that
taxon.  A simple first step is to convert the taxonomic data (for the first record)
into a Pandas DataFrame with</p>
<pre><code class="language-py">import pandas as pd

df = pd.DataFrame(records[0]['taxa'])
</code></pre>
<p>We also need to drop a few lower-level taxa that are already
accounted for in higher ones. e.g. the reads for <em>Bacillus spizizenii TU-B-10</em> are
among the reads counted for <em>Bacillus spizizenii</em>.  A quick way of doing this
is by selecting the rows that have only two words in their names.</p>
<pre><code class="language-py">df = df.loc[df['human_readable'].apply(lambda name: len(name.split()) == 2)]
</code></pre>
<p>Now, let's add columns for the total number of base pairs associated with
each taxon and what proportion that is of the total.</p>
<pre><code class="language-py">df['gDNA'] = df['n_reads']*df['mean_len']
df['proportion'] = df['gDNA']/df['gDNA'].sum()
</code></pre>
<p>Finally, let's make a rough plot with a black dashed line at 12%.</p>
<pre><code class="language-py">import matplotlib.pyplot as plt

plt.plot(df['human_readable'], df['proportion']*100, 'o')
plt.axhline(12, c='k', ls='--');
plt.xticks(rotation=22.5, ha='right');
</code></pre>
<p><a class="glightbox" data-desc-position="bottom" data-height="auto" data-type="image" data-width="auto" href="./zymo-comparison.png"><img alt="Measured gDNA proportions of a Zymo community" src="./zymo-comparison.png"/></a></p>
<p>There are some clear discrepancies—<em>Pseudomonas aeruginosa</em> is
underreported and <em>Bacillus spizizenii</em> is overreported—but this
matches results by e.g. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6520541/">Nicholls et
al. (2019)</a>.</p>
<p>This short example is intended as a basic demonstration of what's
possible in CLIMB-TRE.  We're always interested to hear more examples
of research questions that CLIMB-TRE can answer, so let us know if you
have an example that could be included as a guide for others.</p></div>
</div>
</div><footer>
<hr/>
<div role="contentinfo">
<!-- Copyright etc -->
</div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
</div>
</div>
</section>
</div>
<div aria-label="Versions" class="rst-versions" role="note">
<span class="rst-current-version" data-toggle="rst-current-version">
<span>
<a class="fa fa-github" href="https://github.com/GSTT-CIDR/network_hub" style="color: #fcfcfc"> GitHub</a>
</span>
</span>
</div>
<script src="../js/jquery-3.6.0.min.js"></script>
<script>var base_url = "..";</script>
<script src="../js/theme_extra.js"></script>
<script src="../js/theme.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jvectormap/2.0.5/jquery-jvectormap.js"></script>
<script src="../assets/js/jquery-jvectormap.js"></script>
<script src="../assets/js/jquery-mousewheel.js"></script>
<script src="../assets/js/jvectormap.js"></script>
<script src="../assets/js/abstract-element.js"></script>
<script src="../assets/js/abstract-canvas-element.js"></script>
<script src="../assets/js/abstract-shape-element.js"></script>
<script src="../assets/js/svg-element.js"></script>
<script src="../assets/js/svg-group-element.js"></script>
<script src="../assets/js/svg-canvas-element.js"></script>
<script src="../assets/js/svg-shape-element.js"></script>
<script src="../assets/js/svg-path-element.js"></script>
<script src="../assets/js/svg-circle-element.js"></script>
<script src="../assets/js/svg-image-element.js"></script>
<script src="../assets/js/svg-text-element.js"></script>
<script src="../assets/js/vml-element.js"></script>
<script src="../assets/js/vml-group-element.js"></script>
<script src="../assets/js/vml-canvas-element.js"></script>
<script src="../assets/js/vml-shape-element.js"></script>
<script src="../assets/js/vml-path-element.js"></script>
<script src="../assets/js/vml-circle-element.js"></script>
<script src="../assets/js/vml-image-element.js"></script>
<script src="../assets/js/map-object.js"></script>
<script src="../assets/js/region.js"></script>
<script src="../assets/js/marker.js"></script>
<script src="../assets/js/vector-canvas.js"></script>
<script src="../assets/js/simple-scale.js"></script>
<script src="../assets/js/ordinal-scale.js"></script>
<script src="../assets/js/numeric-scale.js"></script>
<script src="../assets/js/color-scale.js"></script>
<script src="../assets/js/legend.js"></script>
<script src="../assets/js/data-series.js"></script>
<script src="../assets/js/proj.js"></script>
<script src="../assets/js/map.js"></script>
<script src="../assets/js/uk_mill.js"></script>
<script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>
<script>const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});</script></body>
</html>
