<!DOCTYPE html>

<html class="writer-html5" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="IE=edge" http-equiv="X-UA-Compatible"/>
<meta content="width=device-width, initial-scale=1.0" name="viewport"/><meta content="CIDR" name="author"/><link href="https://CIDR-network-hub.github.io/main/running_metagenomics_workflow/" rel="canonical"/>
<link href="../img/favicon.ico" rel="shortcut icon"/>
<title>Running the metagenomics workflow - CIDR Metagenomics Hub</title>
<link href="../css/theme.css" rel="stylesheet"/>
<link href="../css/theme_extra.css" rel="stylesheet"/>
<link href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" rel="stylesheet"/>
<link href="../assets/_mkdocstrings.css" rel="stylesheet"/>
<link href="../assets/js/jquery-jvectormap.css" rel="stylesheet"/>
<link href="../assets/stylesheets/width.css" rel="stylesheet"/>
<link href="https://cdn.datatables.net/2.2.2/css/dataTables.dataTables.min.css" rel="stylesheet"/>
<link href="../css/version-select.css" rel="stylesheet"/>
<script>
        // Current page data
        var mkdocs_page_name = "Running the metagenomics workflow";
        var mkdocs_page_input_path = "running_metagenomics_workflow.md";
        var mkdocs_page_url = "/main/running_metagenomics_workflow/";
      </script>
<!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
<script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
<script>hljs.highlightAll();</script>
<link href="../cidr_metagenomics.pdf" rel="alternate" title="PDF" type="application/pdf"/> <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
    html.glightbox-open { overflow: initial; height: 100%; }
    .gslide-title { margin-top: 0px; user-select: text; }
    .gslide-desc { color: #666; user-select: text; }
    .gslide-image img { background: white; }</style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
<body class="wy-body-for-nav" role="document">
<div class="wy-grid-for-nav">
<nav class="wy-nav-side stickynav" data-toggle="wy-nav-shift">
<div class="wy-side-scroll">
<div class="wy-side-nav-search">
<a class="icon icon-home" href=".."> CIDR Metagenomics Hub
        </a>
</div>
<div aria-label="Navigation menu" class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation">
<p class="caption"><span class="caption-text">Introduction</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="..">Clinical metagenomics</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Lab Resources</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../panmetagenomics_protocol/">Lab protocols</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Bioinformatics</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_outline/">Overview</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../bioinformatics_install/">Installation and validation</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="#">Launching a sequencing experiment using ONT software</a>
<ul>
<li class="toctree-l2"><a class="reference internal" href="../running_minknow/">MinKNOW</a>
</li>
<li class="toctree-l2"><a class="reference internal" href="../running_gourami/">Gourami</a>
</li>
</ul>
</li>
<li class="toctree-l1 current"><a class="reference internal current" href="./">Running the metagenomics workflow</a>
<ul class="current">
<li class="toctree-l2"><a class="reference internal" href="#starting-a-run">Starting a run</a>
</li>
<li class="toctree-l2"><a class="reference internal" href="#feature-descriptions">Feature descriptions:</a>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../running_organism_query/">Query a classification</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../mSCAPE_integration/">Uploading to mSCAPE</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../additional_tools/">Summary report</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../alternative_deployment/">Deployment alternatives</a>
</li>
</ul>
<p class="caption"><span class="caption-text">Analysis</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../clinical_interpretation/">Clinical evaluation SOP</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../validation_outline/">Validation outline</a>
</li>
</ul>
<p class="caption"><span class="caption-text">FAQ</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../faq/">Frequently asked questions</a>
</li>
<li class="toctree-l1"><a class="reference internal" href="../genome_database/">Metagenomics sequence database</a>
</li>
</ul>
</div>
</div>
</nav>
<section class="wy-nav-content-wrap" data-toggle="wy-nav-shift">
<nav aria-label="Mobile navigation menu" class="wy-nav-top" role="navigation">
<i class="fa fa-bars" data-toggle="wy-nav-top"></i>
<a href="..">CIDR Metagenomics Hub</a>
</nav>
<div class="wy-nav-content">
<div class="rst-content"><div aria-label="breadcrumbs navigation" role="navigation">
<ul class="wy-breadcrumbs">
<li><a aria-label="Docs" class="icon icon-home" href=".."></a></li>
<li class="breadcrumb-item">Bioinformatics</li>
<li class="breadcrumb-item active">Running the metagenomics workflow</li>
<li class="wy-breadcrumbs-aside">
<a class="icon icon-github" href="https://github.com/GSTT-CIDR/network_hub/edit/master/docs/running_metagenomics_workflow.md"> Edit on GitHub</a>
</li>
</ul>
<hr/>
</div>
<div class="document" itemscope="itemscope" itemtype="http://schema.org/Article" role="main">
<div class="section" itemprop="articleBody">
<div><h1 id="running-the-cidr-metagenomics-workflow">Running the CIDR metagenomics workflow</h1>
<div class="admonition note">
<p class="admonition-title">Before starting</p>
<ol>
<li>
<p>The CIDR metagenomics workflow must only be executed during a sequencing experiment or after a sequencing experiment has completed. The pipeline must not be launched before a sequencing experiment has been commenced and has <strong>started producing reads</strong>.</p>
</li>
<li>
<p>Ensure the SSD is inserted in to one of the rear USB 3.1 ports, has been mounted and the encryption key has been entered successfully. Test the disk has been mounted by navigating to it in the Ubuntu file explorer.</p>
</li>
</ol>
</div>
<h2 id="starting-a-run">Starting a run</h2>
<ol>
<li>Double click the <strong>Metagenomics Launcher</strong> icon on the GridION desktop, the CIDR Metagenomics Launcher should appear alongside a terminal window.</li>
</ol>
<p><a class="glightbox" data-desc-position="bottom" data-description="Metagenomics launcher 3.8.1" data-height="auto" data-title="GridION desktop screenshot" data-type="image" data-width="auto" href="../img/metagenomics_pipeline_clean.jpg"><img alt="Metagenomics launcher 3.8.1" data-description="Metagenomics launcher 3.8.1" data-title="GridION desktop screenshot" src="../img/metagenomics_pipeline_clean.jpg"/></a></p>
<div class="admonition danger">
<p class="admonition-title">Known issues</p>
<p>The <code>'geocryptfs error not found...'</code> error can be ignored as it is not essential to the workflow.
<br/>
<br/>
For all clinical specimens, positive controls and negatives, a unique Lab/sample ID must be used, taking in to account across all previous runs. For positive controls, for example, add the date to each Lab/sample ID (POS_25_01_18) - you can not use only 'POS' as this will overwrite the previous run's control sample with the same name.</p>
</div>
<h2 id="feature-descriptions">Feature descriptions:</h2>
<table>
<thead>
<tr>
<th>Number</th>
<th>Field</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td><strong>Load existing sample sheet</strong></td>
<td>Load a pre-existing sample sheet. This will populate the fields below with the data from the TSV file.</td>
</tr>
<tr>
<td>2</td>
<td><strong>Number of samples</strong></td>
<td>The number of samples to be analysed. This will create the number of rows in the table below.</td>
</tr>
<tr>
<td>3</td>
<td><strong>Experiment ID</strong></td>
<td>Not to be confused with the ONT</td>
</tr>
<tr>
<td>4</td>
<td><strong>ONT experiment ID</strong></td>
<td>The exact name matching the experiment name on MinKNOW entered by the user when initiating a sequencing run. This is populated automatically from the <code>/data</code> directory.</td>
</tr>
<tr>
<td>5</td>
<td><strong>ONT sample ID</strong></td>
<td>The exact name matching the Sample name on MinKNOW entered by the user when initiating a sequencing run. This is populated automatically from the <code>/data/{experiment_id}/</code></td>
</tr>
<tr>
<td>6</td>
<td><strong>ONT barcode</strong></td>
<td>The ONT library index/barcode used. Green colour indicates the barcode directory has been validated.</td>
</tr>
<tr>
<td>7</td>
<td><strong>Lab/Sample ID</strong></td>
<td>The unique lab accession number for the sample. This data is encrypted before transmission. <strong>If repeating a sample, append with _n</strong></td>
</tr>
<tr>
<td>8</td>
<td><strong>Sample accession</strong></td>
<td>The lab's sample ID - identifying a specific patient specimen - ANONYMISED.</td>
</tr>
<tr>
<td>9</td>
<td><strong>Hospital number</strong></td>
<td>A value identifying the individual providing the sample - ANONYMISED.</td>
</tr>
<tr>
<td>10</td>
<td><strong>Collection date</strong></td>
<td>The date the specimen was collected. For positive and negative controls, this would be the day of library preparation.</td>
</tr>
<tr>
<td>11</td>
<td><strong>Sample Class</strong></td>
<td>The category of the sample loaded.</td>
</tr>
<tr>
<td>12</td>
<td><strong>Sample type</strong></td>
<td>The type of specimen.</td>
</tr>
<tr>
<td>13</td>
<td><strong>Operator</strong></td>
<td>Identifier for user operating the sequencer.</td>
</tr>
<tr>
<td>14</td>
<td><strong>Notes</strong></td>
<td>An open field for notes that will appear on all reports.</td>
</tr>
<tr>
<td>15</td>
<td><strong>Anonymise</strong></td>
<td>Anonymises the 'Sample accession' and 'Hospital number' values using an encryption cypher.</td>
</tr>
<tr>
<td>16</td>
<td><strong>Deanonymise</strong></td>
<td>Deanonymises the 'Sample accession' and 'Hospital number' values present in the launcher fields to their original values. The deanonymisation tool can be used to access previous runs.</td>
</tr>
<tr>
<td>17</td>
<td><strong>Generate Gourami sample sheet</strong></td>
<td><strong>Only for Q-line &gt;=v1.1</strong> Generates a Gourami compatible sample sheet for starting a sequencing experiment. The output can be found in the <code>./metagenomics/sample_Sheet/gourami</code> directory.</td>
</tr>
<tr>
<td>18</td>
<td><strong>Force overwrite</strong></td>
<td>Checking this box will move results and reports for all timepoints matching the 'Lab/sample ID' filed in the the launcher to the <code>./metagenomics/recycle_bin</code> directory and 'unlock' all directories. If you have aborted a run, or the terminal is reporting failures, try using this feature.</td>
</tr>
<tr>
<td>19</td>
<td><strong>mSCAPE prompt</strong></td>
<td>After the sequencing and analysis run has completed, open the mSCAPE uploader for user input. No data is uploaded without par-sample expressed authorisation.</td>
</tr>
<tr>
<td>20</td>
<td><strong>Select timepoints</strong></td>
<td>Select the timepoints you'd like to be generated. If you encounter errors generating a timepoint visit the <a href="../faq/">FAQ section</a></td>
</tr>
<tr>
<td>21</td>
<td><strong>Refresh directories</strong></td>
<td>This button refreshes the contents of the MinKNOW experiment ID and MinKNOW sample ID columns. Useful if you have started the launcher before commencing the sequencing experiment.</td>
</tr>
<tr>
<td>22</td>
<td><strong>Launch pipeline</strong></td>
<td>Launches metagenomics analysis, saving the sample sheet to the <code>./metagenomics/sample_sheets</code>.</td>
</tr>
</tbody>
</table>
<ol start="2">
<li>
<p>Select the number of samples to be analysed from the dropdown (Feature 2).</p>
</li>
<li>
<p>Add an appropriate experiment ID to the field (Feature 3). This will label the stored sample sheet, useful for auditing. For Gourami users, this will set the ONT experiment ID.</p>
</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul>
<li>Launching an analysis run will save all of the data in the launcher fields to a TSV file in the <code>metagenomics/sample_sheets</code> directory, with the date/time and the contents of the Experiment ID field. This can be reused if a repeat run is required - or quick edits need to be made to a set of samples without having to fill out the fields again.</li>
</ul>
</div>
<ol start="4">
<li>
<p>Select the experiment and sample ID corresponding to the MinKNOW/Gourami run from the ONT Experiment ID and ONT sample ID dropdown menu (Feature 4). If your experiment is not there, click the refresh button (Feature 21).</p>
</li>
<li>
<p>Complete the remaining fields using the table above as a guide. The NHS service evaluation and mSCAPE protocols require that all fields are completed where appropriate.</p>
</li>
<li>
<p>Where appropriate, apply the anonymisation functions to the data using the 'Anonymise' button. This will pseudo-anonymise the 'Sample accession' and 'Hospital number' fields. Users can deanonymise a sample sheet by loading it in to the lanucher and selecting 'Deanonymise' or by using the <a href="./deanonymisation.md">Deanonymisation tool</a>. </p>
</li>
<li>
<p>Click on <code>Launch pipeline</code> and follow the instructions to start the analysis.</p>
</li>
<li>
<p>After a minute, the terminal window accompanying the workflow launcher should start displaying log outputs from the workflow. See below for an example.</p>
</li>
<li>
<p>~35 minutes after launching the sequencing experiment alongside the metagenomics workflow, the first reports will be available in <code>/media/grid/metagenomics/reports/{sample_name}/</code>. See below for a guide on how to access this.</p>
</li>
</ol>
<div class="admonition tip">
<p class="admonition-title">Success!</p>
<p>The workflow will run for ~24 hours generating reports for 0.5, 1, 2, 16, 24 hour time-points. If you are using a previously sequenced dataset, reports will be generated as quickly as possible.</p>
</div></div>
</div>
</div><footer>
<div aria-label="Footer Navigation" class="rst-footer-buttons" role="navigation">
<a class="btn btn-neutral float-left" href="../running_gourami/" title="Gourami"><span class="icon icon-circle-arrow-left"></span> Previous</a>
<a class="btn btn-neutral float-right" href="../running_organism_query/" title="Query a classification">Next <span class="icon icon-circle-arrow-right"></span></a>
</div>
<hr/>
<div role="contentinfo">
<!-- Copyright etc -->
</div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
</div>
</div>
</section>
</div>
<div aria-label="Versions" class="rst-versions" role="note">
<span class="rst-current-version" data-toggle="rst-current-version">
<span>
<a class="fa fa-github" href="https://github.com/GSTT-CIDR/network_hub" style="color: #fcfcfc"> GitHub</a>
</span>
<span><a href="../running_gourami/" style="color: #fcfcfc">« Previous</a></span>
<span><a href="../running_organism_query/" style="color: #fcfcfc">Next »</a></span>
</span>
</div>
<script src="../js/jquery-3.6.0.min.js"></script>
<script>var base_url = "..";</script>
<script src="../js/theme_extra.js"></script>
<script src="../js/theme.js"></script>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.7.1/jquery.min.js"></script>
<script src="https://cdn.datatables.net/2.2.2/js/dataTables.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jvectormap/2.0.5/jquery-jvectormap.js"></script>
<script src="../assets/js/jquery-jvectormap.js"></script>
<script src="../assets/js/jquery-mousewheel.js"></script>
<script src="../assets/js/jvectormap.js"></script>
<script src="../assets/js/abstract-element.js"></script>
<script src="../assets/js/abstract-canvas-element.js"></script>
<script src="../assets/js/abstract-shape-element.js"></script>
<script src="../assets/js/svg-element.js"></script>
<script src="../assets/js/svg-group-element.js"></script>
<script src="../assets/js/svg-canvas-element.js"></script>
<script src="../assets/js/svg-shape-element.js"></script>
<script src="../assets/js/svg-path-element.js"></script>
<script src="../assets/js/svg-circle-element.js"></script>
<script src="../assets/js/svg-image-element.js"></script>
<script src="../assets/js/svg-text-element.js"></script>
<script src="../assets/js/vml-element.js"></script>
<script src="../assets/js/vml-group-element.js"></script>
<script src="../assets/js/vml-canvas-element.js"></script>
<script src="../assets/js/vml-shape-element.js"></script>
<script src="../assets/js/vml-path-element.js"></script>
<script src="../assets/js/vml-circle-element.js"></script>
<script src="../assets/js/vml-image-element.js"></script>
<script src="../assets/js/map-object.js"></script>
<script src="../assets/js/region.js"></script>
<script src="../assets/js/marker.js"></script>
<script src="../assets/js/vector-canvas.js"></script>
<script src="../assets/js/simple-scale.js"></script>
<script src="../assets/js/ordinal-scale.js"></script>
<script src="../assets/js/numeric-scale.js"></script>
<script src="../assets/js/color-scale.js"></script>
<script src="../assets/js/legend.js"></script>
<script src="../assets/js/data-series.js"></script>
<script src="../assets/js/proj.js"></script>
<script src="../assets/js/map.js"></script>
<script src="../assets/js/uk_mill.js"></script>
<script src="../js/version-select.js"></script>
<script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>
<script>const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});</script></body>
</html>
