# Clinical metagenomics bioinformatics

## Overview
The principal deliverable of the clinical metagenomics service at GSTT is a report describing all organisms present within a sample with detectable nucleic acids. The report is generated by the CIDR Metagenomics Pipeline, intepreted by a bioinformatician a clinical scientist/microbiologist and a clinician and passed to Epic in a report similar to that which one would receive from microbiology. 

### Technical facets
After loading a library on to an ONT sequencing device and launching the sequencing experiment in ONT MinKNOW, the pipeline is initialised by the user, where it periodically ingests base called FASTQ data, analyses and reports at set intervals - 0.5, 1, 2, 16 and 24 hours. The pipeline performs human scrubbing, taxanomic classification, AMR detection and MLST which is then consolidated in to a PDF report.

#### Taxanomic classification
At its core, the pipeline depends of the process of taxanomic classification, for which there are many tools. Presently we use [Centrifuge](https://ccb.jhu.edu/software/centrifuge/manual.shtml). 